pactflow/client.ts

```diff
@@ -143,4 +162,20 @@ export class PactflowClient implements Client {
         );
       }
     }
+
+    // Register resources for supported clients.
+    // To add a new resource:
+    //   1. Define the list of allowed clients in the `if` condition.
+    //   2. Call `register` with the resource name, path, and handler.
+    //   3. Only clients in the allowed list will have access to that resource.
+    registerResources(register: RegisterResourceFunction): void {
+      if(["pactflow", "pact_broker"].includes(this.clientType)) {
+        register("get-provider-states", "{provider}", async (uri, variables, _extra) => {
+          const states = await this.getProviderStates(variables.provider as string);
+          return {
+            contents: [{ uri: uri.href, text: JSON.stringify(states) }]
```

Comment:
Should we create a type for this response, to help the agent know what to expect here?